Apps with Typescript <5 can't compile this library

My app that is still on typescript 4 gets compiler errors when trying to use this library. Upgrading to typescript 5 fixes these errors but isn't feasible for the time being.

```
node_modules/@praha/byethrow/dist/cjs/functions/combine.d.ts:51:6 - error TS1139: Type parameter declaration expected.
51     <const X extends Array<ResultMaybeAsync<any, any>>>(x: X): ResultFor<X[number], {
        ~~~~~
node_modules/@praha/byethrow/dist/cjs/functions/combine.d.ts:51:14 - error TS1005: ',' expected.
51     <const X extends Array<ResultMaybeAsync<any, any>>>(x: X): ResultFor<X[number], {
                ~~~~~~~
node_modules/@praha/byethrow/dist/cjs/functions/combine.d.ts:51:27 - error TS1005: ',' expected.
51     <const X extends Array<ResultMaybeAsync<any, any>>>(x: X): ResultFor<X[number], {
node_modules/@praha/byethrow/dist/cjs/functions/combine.d.ts:51:55 - error TS1109: Expression expected.
51     <const X extends Array<ResultMaybeAsync<any, any>>>(x: X): ResultFor<X[number], {
node_modules/@praha/byethrow/dist/cjs/functions/combine.d.ts:51:58 - error TS1005: ')' expected.
51     <const X extends Array<ResultMaybeAsync<any, any>>>(x: X): ResultFor<X[number], {
```

--- a/.changeset/honest-buckets-stop.md
+++ b/.changeset/honest-buckets-stop.md
@@ -1,0 +1,5 @@
+---
+"@praha/byethrow": patch
+---
+
+Make it require TypeScript v5 or higher

--- a/packages/byethrow/package.json
+++ b/packages/byethrow/package.json
@@ -59,6 +59,9 @@
     "typescript": "5.9.3",
     "vitest": "3.2.4"
   },
+  "peerDependencies": {
+    "typescript": ">=5.0.0"
+  },
   "publishConfig": {
     "provenance": true
   }

